### PR TITLE
feat: remove scripts section in package.json

### LINKED ISSUE #662: [Feature] Remove scripts section from distribution-ready package.json
## Type of Issue

```
[ ] Bug Report
[x] Feature Request
```

## Description

Option to remove scripts from `package.json` while building package. So the outputted `package.json` won't have scripts section. 

Motivation: http://blog.npmjs.org/post/141702881055/package-install-scripts-vulnerability.

#### Expected Behaviour

remove scripts from final package.json

```js
{
    "$schema": "./node_modules/ng-packagr/ng-package.schema.json",
    "workingDirectory": "./.ng_pkg_build",
    "dest": "./build/",
    "lib": {
        "entryFile": "./src/public_api.ts",
        "ignoreScripts": true //expected
    }
}

```

### PR DESCRIPTION
**Pull request by Sibiraj-S**
closes #662

## I'm submitting a...

```
[ ] Bug Fix
[x] Feature
[ ] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [x] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [x] Tests for the changes have been added

## Description

Ignore package.json scripts while bundling

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/ng-package.schema.json [hunk 1]
```diff
@@ -98,6 +98,11 @@
             "type": "string"
           },
           "default": ["dom", "es2015"]
+        },
+        "ignoreScripts": {
+          "description": "Remove scripts from package.json after build",
+          "type": "boolean",
+          "default": false
```
Comment (dherges) [inline src/ng-package.schema.json]: Ok, from the npm blog post, I don't see a motivation why a library package would need to include lifecycle scripts. I would either

- make this option `true` by default, making it opt-in to publish lifecycle scripts
- or: make the `delete packageJson.scripts` behaviour turned on by default and non-overwriteable

Comment (dherges) [reply src/ng-package.schema.json]: And maybe name it `keepLifecycleScripts`?

Comment (Sibiraj-S) [reply src/ng-package.schema.json]: sure.

Comment (Sibiraj-S) [reply src/ng-package.schema.json]: I kept that false, since I don't want to change the default behavior how ng-packagr works

**Comment by Sibiraj-S** [comment]
Created a new PR #684, since the master branch was deleted.

**Pull request closed without merging**